src/main/java/org/sobotics/guttenberg/commands/Update.java

```diff
@@ -0,0 +1,54 @@
+package org.sobotics.guttenberg.commands;
+
+import org.sobotics.guttenberg.clients.Updater;
+import org.sobotics.guttenberg.utils.CommandUtils;
+
+import fr.tunaki.stackoverflow.chat.Message;
+import fr.tunaki.stackoverflow.chat.Room;
+
+public class Update implements SpecialCommand {
+
+	private Message message;
+
+    public Update(Message message) {
+        this.message = message;
+    }
+	
+    @Override
+	public boolean validate() {
+		return CommandUtils.checkForCommand(message.getPlainContent(),"update");
+	}
+
+	@Override
+	public void execute(Room room) {
+		System.out.println("Load updater...");
+		Updater updater = new Updater();
+		System.out.println("Check for updates...");
+		
+		int update = updater.updateIfAvailable(); 
+		
+		if (update == -1) {
+			room.send("Automatic update failed!");
```

Comment:
Instead of returning -1 in case of a failed update, throw an exception. Then you can catch it at the right place so as not to duplicate the code handling `update == -1`.